static/js/index.js

```diff
@@ -114,26 +114,32 @@ class App extends React.Component {
         })
     }
     toggleTag(e) {
-        var tags = this.state.tags;
-        tags[e.target.value] = !tags[e.target.value];
+        const tagName = e.target.value;
+        // Make a copy of the tags in state (we shouldn't update the array
+        // in the state, otherwise React won't rerender our components)
+        const tags = [...this.state.tags];
+        tags[tagName] = !tags[tagName];
```

Comment:
Will modifying a key of an object work if the object is declared with `const`? Doesn't it need to be `var`? I thought `const` variables were immutable.

Reply:
If a variable is declared to be `const`, then you just can't assign a new value to that variable. It does not prevent you from modifying the object or array (not that we're even doing that here).

Also, apparently `const` is allowable in a for-each loop because it is just redeclared every time through the loop. Or at least that's what I interpret it to be doing. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
huh nice